internal/kafka/internal/services/quota/ams_quota_service_test.go

```diff
@@ -2390,130 +2331,3 @@ func Test_amsQuotaService_IsQuotaEntitlementActive(t *testing.T) {
 		})
 	}
 }
-
-func Test_amsQuotaService_getAMSRelatedResourceBillingModel(t *testing.T) {
```

Comment:
I think we should still have a unit test for this refactored method to ensure we get the correct billing model.

Reply:
The method has been deleted. We don't call `reserved resources` anymore.

Reply:
but it's been refactored to `amsBillingModelToAMSBillingModelCategory` which should have its own unit test. I think this can be refactored to accommodate for the changes there or create a new one.

Reply:
It has not been refactored in that method. It is a totally different method.
`amsBillingModelToAMSBillingModelCategory` just returns `marketplace` for any marketplace ams billing model and `standard` for `standard`.

Reply:
What Jameel is trying to say is that we should have unit tests for it. It is a new method.

Reply:
I see. I'm not a fan of tests on private interface, other than that the method is as simple as a switch on 3 values. I don't thin it worths a test. In any case, if we want, I agree with @JameelB in adding it in #1549

Reply:
> I see. I'm not a fan of tests on private interface, other than that the method is as simple as a switch on 3 values. I don't thin it worths a test. In any case, if we want, I agree with @JameelB in adding it in #1549

As a side comment, I think the answer to whether testing private interface is good or not is "it depends". It is not as simple as "private tests shouldn't be tested".
Sometimes the public API of a type is very broad and then it is reasonable to test private methods. Of course, if the public API of a type is very broad this is a potential symptom of a design issue where the class "doing too much". For that the alternative is having smaller types that do more specific things.
But, as we know that's not the code we have / can control /change in a reasonable time in some cases, and even the small types might be package-private so the same question would apply.